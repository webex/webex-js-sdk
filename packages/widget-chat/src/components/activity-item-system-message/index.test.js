import React from 'react';
import renderer from 'react-test-renderer';

import ActivityItemSystemMessage from '.';

describe(`ActivityItemSystemMessage tombstone component`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `tombstone`
  };

  const component = renderer.create(
    <ActivityItemSystemMessage
      id={activity.id}
      name={activity.name}
      timestamp={activity.timestamp}
      verb={activity.verb}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});

describe(`ActivityItemSystemMessage tombstone component is self`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `tombstone`
  };

  const component = renderer.create(
    <ActivityItemSystemMessage
      id={activity.id}
      isSelf
      name={activity.name}
      timestamp={activity.timestamp}
      verb={activity.verb}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});

describe(`ActivityItemSystemMessage create component`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `create`
  };

  const component = renderer.create(
    <ActivityItemSystemMessage
      id={activity.id}
      name={activity.name}
      timestamp={activity.timestamp}
      verb={activity.verb}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});

