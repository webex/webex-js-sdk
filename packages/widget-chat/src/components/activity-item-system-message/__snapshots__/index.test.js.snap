exports[`ActivityItemSystemMessage create component renders properly 1`] = `
<div
  className="system-message">
  Test User created conversation. 2016-09-20T19:52:57.186Z
</div>
`;

exports[`ActivityItemSystemMessage tombstone component is self renders properly 1`] = `
<div
  className="system-message">
  You deleted your message. 2016-09-20T19:52:57.186Z
</div>
`;

exports[`ActivityItemSystemMessage tombstone component renders properly 1`] = `
<div
  className="system-message">
  Test User deleted their own message. 2016-09-20T19:52:57.186Z
</div>
`;
